pkg/aggregator/time_sampler.go

```diff
@@ -20,6 +23,9 @@ type SerieSignature struct {
 	nameSuffix string
 }
 
+// TimeSamplerID is a type ID for sharded time samplers.
+type TimeSamplerID int
+
 // TimeSampler aggregates metrics by buckets of 'interval' seconds
 type TimeSampler struct {
 	interval                    int64
```

Comment:
While I agree it has grew now that it is responsible of its own flushing, the `TimeSampler` has always be responsible of creating the sketches and series since it is... sampling them in time, so it is part of its responsibility.
As mention in the code though, the added code introduced to serialize the "stream series" since you introduced your change will have to be refactored with the Aggregator / other samplers at some point, but no reason to do it now afaik, especially if we want to keep this pull-request reviewable.

Reply:
Created the `timeSamplerWorker` to address this.